src/darwin/Framework/CHIP/MTRCommissionableBrowser.mm

```diff
@@ -109,7 +113,14 @@ CHIP_ERROR Stop()
         mDiscoveredResults = nil;
 
 #if CONFIG_NETWORK_LAYER_BLE
-        ReturnErrorOnFailure(PlatformMgrImpl().StopBleScan());
+        mBleScannerDelegateOwner = owner; // retain the owner until OnBleScanStopped is called
+        CHIP_ERROR err = PlatformMgrImpl().StopBleScan();
+        if (err != CHIP_NO_ERROR) {
+            mBleScannerDelegateOwner = nil;
```

Comment:
I think we should just leave mBleScannerDelegateOwner in this case, with a comment that a leak is better than use-after-free...

Reply:
Or I guess we could document why if StopBleScan fails we can assume there is in fact no scan and hence that no one will try calling us.